### ISSUE TITLE
Unexpected infinite loop with `Invoke`

### ISSUE THREAD

**Issue by hajimehoshi** [gopherjs/gopherjs#819]
```go
package main

import (
        "github.com/gopherjs/gopherjs/js"
)

func main() {
        ch := make(chan int)
        c := js.MakeFunc(func(this *js.Object, args []*js.Object) interface{} {
                ch <- args[0].Int() + args[1].Int()
                return nil
        })
        c.Invoke(1, 2)
        println(<-ch)
}
```

shows

```
3
3
3
3
3
3
3
...
```

**Comment by myitcv** [comment]
I don't think this is a bug; because the callback may block you have to do the blocking operation in a go routine: https://github.com/gopherjs/gopherjs#goroutines

**Comment by hajimehoshi** [comment]
Hmm, I understand goroutine is required, but but is this correct to get the infinite loop? Should there be an exception?

**Comment by myitcv** [comment]
How would you detect this?

**Comment by hajimehoshi** [comment]
I have no idea: I'm not familiar with goroutine implementation in GopherJS...

**Comment by myitcv** [comment]
As far as I've thought about this before, I think it would be hard to detect in the compiler, and I can't see how you would do this at runtime either.

**Comment by myitcv** [comment]
> Is there somehow a bug in how GopherJS goroutine scheduling works? 

I don't think so. I think it's more a case that the behaviour is undefined if you don't use a go routine or buffered channel as you suggest.
